bin/runtime/src/lib.rs

```diff
@@ -344,13 +344,109 @@ parameter_types! {
     pub const Offset: u32 = 0;
 }
 
+fn rotate<T: Clone + PartialEq>(
+    current_era: EraIndex,
+    current_session: SessionIndex,
+    n_validators: usize,
+    mut all_validators: Vec<T>,
+    mut reserved: Vec<T>,
+) -> Option<Vec<T>> {
+    if current_era == 0 {
+        return None;
+    }
+
+    all_validators.retain(|v| !reserved.contains(v));
```

Comment:
What I dislike about this function is the overloading of names because of mutability.
Here `all_validators` won't actually contain all of the validators and at the end of the function the `reserved` vector won't contain only the reserved nodes. Seems to me that since `T: Clone` anyway, we could do this without making `all_validators` and `reserved` mutable (i.e. use `filter` instead of `retain` and so on).

That said, this is still quite a nice readable piece of code, so I'm just leaving this here for future consideration 🙂

Reply:
done